compiler/z/codegen/UnaryEvaluator.cpp

```diff
@@ -520,9 +520,14 @@ OMR::Z::TreeEvaluator::l2aEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    node->setRegister(source);
 
 
-   //printf("Node %p is first child\n", firstChild); fflush(stdout);
-   //traceMsg(comp, "Node %p is first child\n", firstChild);
-   cg->decReferenceCount(firstChild);
+   if (shouldGenerateGuardedLoadsForCompressedLoad)
```

Comment:
Good catch. I will add TR::Compiler->om.compressedReferenceShift() == 0 to the check